src/ParticleArrayHelper.cpp

```diff
@@ -0,0 +1,30 @@
+#include "ParticleArrayHelper.hpp"
+
+#include <stdexcept>
+
+
+
+double * ParticleArrayHelper::getParticleArray(double * x, const int &i){
+    return x+_nspacedim*i;
+}
+
+
+const double * ParticleArrayHelper::getParticleArray(const double * x, const int &i){
+    return x+_nspacedim*i;
+}
+
+
+void ParticleArrayHelper::setParticleArray(double * x, const int &i, const double * newx){
+    const int istart = _nspacedim*i;
+    for (int k=0; k<_nspacedim; ++k){
+        x[istart + k] = newx[k];
+    }
+}
+
+
+void ParticleArrayHelper::addArrayToParticleArray(double * x, const int &i, const double * toadd){
+    const int istart = _nspacedim*i;
+    for (int k=0; k<_nspacedim; ++k){
+        x[istart + k] += toadd[k];
+    }
+}
```

Comment:
What is the reason to put a matrix in 1D array and then having to use such a helper?

Reply:
Well, the reason is that the MCI library works like this...
And, as a consequence, this is how VMC works as well: the particle position is a double *.
Then this helper it's nice for making things a little bit easier